diff -uNar drupal-6.8/includes/database.pgsql.inc drupal-6.8.new/includes/database.pgsql.inc
--- drupal-6.8/includes/database.pgsql.inc	2008-11-17 04:31:41.000000000 -0600
+++ drupal-6.8.new/includes/database.pgsql.inc	2008-12-17 15:40:14.000000000 -0600
@@ -341,7 +341,7 @@
  *  Encoded data.
  */
 function db_encode_blob($data) {
-  return "'". pg_escape_bytea($data) ."'";
+  return "E'". pg_escape_bytea($data) ."'";
 }
 
 /**
