From a2c3ff8be8373aad2650cd0d11cd0667d0348157 Mon Sep 17 00:00:00 2001
From: Clayton Smith <argilo@gmail.com>
Date: Sun, 7 Jun 2020 16:12:39 -0400
Subject: [PATCH] Fix PortAudio with GNU Radio 3.8

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f86c3575..fe35fd11 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,7 +147,11 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
         unset(PORTAUDIO_INCLUDE_DIRS CACHE)
         unset(PORTAUDIO_LIBRARIES CACHE)
     elseif(${LINUX_AUDIO_BACKEND} MATCHES "Portaudio")
-        find_package(Portaudio REQUIRED)
+        if(Gnuradio_VERSION VERSION_LESS "3.8")
+            find_package(Portaudio REQUIRED)
+        else()
+            find_package(PORTAUDIO REQUIRED)
+        endif()
         add_definitions(-DWITH_PORTAUDIO)
         unset(PULSEAUDIO_FOUND CACHE)
         unset(PULSEAUDIO_INCLUDE_DIR CACHE)
